eth_pydantic_types/abi.py

```diff
@@ -0,0 +1,105 @@
+"""
+These models are used to match the lowercase type names used by the abi.
+"""
+
+from typing import ClassVar
+
+from pydantic_core.core_schema import CoreSchema, str_schema
+
+from .address import Address
+from .hex import _make_cls as _make_hex_cls
+from .numbers import _make_cls as _make_number_cls
+
+bytes1 = _make_hex_cls(1, bytes, dict_additions=dict(abi_type="bytes1"))
+bytes4 = _make_hex_cls(4, bytes, dict_additions=dict(abi_type="bytes4"))
+bytes8 = _make_hex_cls(8, bytes, dict_additions=dict(abi_type="bytes8"))
+bytes16 = _make_hex_cls(16, bytes, dict_additions=dict(abi_type="bytes16"))
+bytes20 = _make_hex_cls(20, bytes, dict_additions=dict(abi_type="bytes20"))
+bytes32 = _make_hex_cls(32, bytes, dict_additions=dict(abi_type="bytes32"))
+bytes64 = _make_hex_cls(64, bytes, dict_additions=dict(abi_type="bytes64"))
+
+
+class string(str):
+    """
+    Represents a string.
+    """
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, value, handler=None) -> CoreSchema:
+        return str_schema()
+
+    abi_type: ClassVar[str] = "string"
+
+
+class address(Address):
+    """
+    Represents a 20 character hex string address.
+    """
+
+    abi_type: ClassVar[str] = "address"
+
+
+int8 = _make_number_cls(8, int, dict_additions=dict(abi_type="int8"))
```

Comment:
we should make a note to refactor these to class definitions in the future.
from the research and findings, making dynamic classes should only be done if absolutely necessary.
in our case, we have the ability to have all these defined at the beginning of runtime (rather than during runtime) and that affords us a lot more stuff, such as the ability to use TypeAlias and NewType, but maybe even more I don't realize..

**This is a future, don't worry about this now, but maybe we should track the issue**